validator/project_validator.go

```diff
@@ -637,6 +637,9 @@ func ensureReferentialIntegrity(project *model.Project, containerNameMap map[str
 	for _, buildVariant := range project.BuildVariants {
 		buildVariantTasks := map[string]bool{}
 		for _, task := range buildVariant.Tasks {
+			if task.Group != nil {
```

Comment:
We should check all the tasks in the task group like we do at 630 - 633 i think.